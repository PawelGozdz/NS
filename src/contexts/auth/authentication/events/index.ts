export * from './on-user-updated/on-user-updated.handler';
