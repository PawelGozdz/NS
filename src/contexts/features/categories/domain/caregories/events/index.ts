export * from './category-created.event';
export * from './category-updated.event';
