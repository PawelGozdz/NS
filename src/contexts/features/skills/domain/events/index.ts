export * from './skill-created.event';
