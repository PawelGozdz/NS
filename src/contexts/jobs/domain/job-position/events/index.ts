export * from './job-position-created.event';
export * from './job-position-updated.event';
