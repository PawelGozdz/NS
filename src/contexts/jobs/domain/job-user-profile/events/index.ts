export * from './job-user-profile-created.event';
export * from './job-user-profile-updated.event';
