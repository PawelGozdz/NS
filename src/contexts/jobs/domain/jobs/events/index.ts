export * from './job-created.event';
export * from './job-updated.event';
