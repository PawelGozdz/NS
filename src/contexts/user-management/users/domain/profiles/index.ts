export * from './errors';
export * from './profile.entity';
export * from './profile.snapshot';
export * from './profiles-command-repository.interface';
export * from './profiles-query-repository.interface';
