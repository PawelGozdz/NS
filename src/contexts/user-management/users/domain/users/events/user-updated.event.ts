import { Actor, Address } from '@app/core';
import { EntityId } from '@libs/common';
import { Event } from '@libs/cqrs';

export class UserUpdatedEvent extends Event<UserUpdatedEvent> {
  id: EntityId;

  email: string;

  profile: {
    id: EntityId;
    userId: EntityId;
    firstName: string | null;
    lastName: string | null;
    username: string | null;
    address: Address | null;
    bio: string | null;
    dateOfBirth: Date | null;
    gender: string | null;
    hobbies: string[];
    languages: string[];
    phoneNumber: string | null;
    profilePicture: string | null;
    rodoAcceptanceDate: Date | null;
  };

  actor: Actor;

  constructor(event: UserUpdatedEvent) {
    super(event);

    Object.assign(this, event);
  }
}
