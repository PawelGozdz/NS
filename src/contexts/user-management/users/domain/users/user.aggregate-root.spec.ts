import { Actor } from '@app/core';
import { ActorType, EntityId } from '@libs/common';

import { Profile } from '../profiles';
import { UserCreatedEvent, UserUpdatedEvent } from './events';
import { User } from './user.aggregate-root';
import { UserAggregateRootFixtureFactory } from './user.aggregate-root.fixture';

const email = 'test@test.pl';

describe('User', () => {
  afterEach(() => {
    jest.resetModules();
    jest.restoreAllMocks();
  });

  const actor = Actor.create(ActorType.USER, User.name, '34d90467-aeb4-4016-9791-86f9aec010e0');

  describe('create', () => {
    describe('Success', () => {
      it('should apply UserCreatedEvent with property values', () => {
        // arrange

        // act
        const user = User.create({
          email,
          profile: {
            id: EntityId.createRandom(),
            userId: EntityId.createRandom(),
            firstName: 'test',
          },
          actor,
        });

        // assert
        const events = user.getUncommittedEvents();

        expect(events.length).toEqual(1);
        expect(events[0]).toBeInstanceOf(UserCreatedEvent);

        expect(events[0]).toMatchSnapshot({
          id: expect.any(EntityId),
          profile: new Profile({
            firstName: 'test',
            id: expect.any(EntityId),
            userId: expect.any(EntityId),
          }),
        });
      });
    });
  });

  describe('update', () => {
    it('should apply UserUpdatedEvent with properties to be updated', () => {
      // arrange
      const user = UserAggregateRootFixtureFactory.create();

      const newEmail = 'new@test.pl';

      // act
      user.update({ email: newEmail, actor });

      // assert
      const events = user.getUncommittedEvents();

      expect(events[0]).toBeInstanceOf(UserUpdatedEvent);
      expect(user.getEmail()).toEqual(newEmail);
    });
  });
});
