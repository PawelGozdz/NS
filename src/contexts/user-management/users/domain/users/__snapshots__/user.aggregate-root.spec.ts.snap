// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User create Success should apply UserCreatedEvent with property values 1`] = `
{
  "actor": Actor {
    "id": "34d90467-aeb4-4016-9791-86f9aec010e0",
    "source": "User",
    "type": "user",
  },
  "email": "test@test.pl",
  "id": Any<EntityId>,
  "profile": {
    "address": null,
    "bio": null,
    "dateOfBirth": null,
    "firstName": "test",
    "gender": null,
    "hobbies": [],
    "id": Any<EntityId>,
    "languages": [],
    "lastName": null,
    "phoneNumber": null,
    "profilePicture": null,
    "rodoAcceptanceDate": null,
    "userId": Any<EntityId>,
    "username": null,
  },
}
`;
